contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,499 @@
+---
```

Comment:
Some general comments:
- Let us mention explicitly the role that UpgradeTask plays
- Let us clearly mention who can create UpgradeTask
- Let us clearly mention who makes use of UpgradeTask
- Let us not link resources via their names rather prefer use of labels & labelSelector
- If UpgradeTask can also act as a creator of a Job then UpgradeTask should embed the JobTemplate/PodTemplate in some form
- If job/binary can update the UpgradeTask, then these updates should be done to a separate sub resource
- Let us debate if we should move complex schema of status to a sub resource
- IMO UpgradeTask can be user facing / residing in user's git repo or can be internal to other higher order tools. So let us figure out the `apply semantics` / challenges as much as possible.

Reply:
Addressed most of the comments except for the schema changes. As part of the 1.2 release, another OEP update will be raised to flush out the details of the schema.